### ISSUE TITLE
3.0 - Consider forcing the order_number column to be unique

### ISSUE THREAD

**Issue by robincornett** [awesomemotive/easy-digital-downloads#9068]
In thinking about assigning custom order numbers to orders in EDD 3.0, it seems to me that we should consider setting a unique constraint on the `order_number` column in the orders table. Similar to the ID property, I am not sure there is a scenario where it would be acceptable to have two orders with identical `order_number` properties, for orders with order numbers.

**Comment by JJJ** [comment]
👋 @robincornett – I see what you mean, and I agree that preventing dupes there seems correct.

In my experience, I've come to prefer having the `PRIMARY KEY` as the only unique column in the database table schema, and then writing methods in PHP to restrict things as needed, though there are pros and cons either way.

There is a high likelihood that EDD will want/need to check if an `order_number` already exists, and that same method/approach would be used to avoid a duplicate entry.

Not using the MySQL constraint also allows for the future option of reusing an `order_number` – maybe for a revision or something.

Enforcing this also requires some checking and decision-making on what to do when duplicate order numbers already exist in the database. MySQL would just error, but doing it in PHP allows you to alert the user to reconcile the problem.

**Comment by cklosowski** [comment]
I agree with @JJJ here, if we decide to go this route (we need to do some more testing and planning if we do), we should do the restrictions in PHP not at the MySQL level.
